src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -1,236 +1,637 @@
 /*
  * File: z_boss_01.c
  * Overlay: ovl_Boss_01
- * Description: Odolwa
+ * Description: Odolwa, his bugs, and his afterimages.
+ *
+ * In addition to handling Odolwa, this actor is also responsible for handling the bugs that Odolwa can spawn as well as
+ * the afterimages that appear when Odolwa jumps or does a spin attack. The bugs are effectively their own actor with
+ * their own Update and Draw functions and a whole dedicated section of the file specifically for them. Afterimages are
+ * much simpler, and their code is more interleaved with Odolwa's code. This actor also handles the falling block and
+ * ring of fire effects that Odolwa can summon.
+ *
+ * Odolwa's behavior can primarily be divided into two states: waiting and attacking. He has a variety of wait actions
+ * and attack types that he can select between, and this can make it difficult to tell when he is waiting and when he is
+ * attacking. Two of his wait actions look like attacks, and when selecting an attack, he can randomly choose to dance
+ * for a bit before attacking. Additionally, Odolwa can attack the player via more indirect means while waiting by
+ * summoning bugs, by dropping falling blocks on them, or by summoning a ring of fire to surround them.
+ *
+ * Outside of waiting and attacking, most of Odolwa's behavior involves either dodging attacks (e.g., by jumping,
+ * running around, or guarding) or reacting to attacks. He also handles all his cutscenes (his intro cutscene, the
+ * cutscene where he summons bugs, and his death cutscene) manually via actionFuncs.
+ *
+ * Odolwa's fight can be divided roughly into two phases, but unlike most bosses, the second phase does not happen when
+ * his health reaches a certain point. Rather, the second phase starts when 1000 or more frames have passed since the
+ * fight began. At this point, he will play the bug summoning cutscene, and more wait actions will be available for him
+ * to randomly select; these new wait actions are what allow him to summon a ring of fire, drop falling blocks, etc.
```

Comment:
Sort of. After 1000 frames, it will enter the second phase the next time it calls SetupWait. And assuming the game consistently runs at 20 frames per second, that will be around 50 seconds into the fight. I prefer writing things out in terms of frames, not seconds, though, because this game does not consistently run at 20 FPS.